### ISSUE TITLE
ApolloServerPluginUsageReporting does not allow to pass http agent

### ISSUE THREAD

**Issue by vany0114** [apollographql/federation#2897]
### Issue Description

Hello,

I'm using Apollo Federation in an Azure environment, specifically deployed as an Azure Function. I've noticed that the `ApolloServerPluginUsageReporting` is causing a lot of `connect ETIMEDOUT` as it seems it is exhausting the available Websockets/TCP connections when making this request to: `HTTP: POST usage-reporting.api.apollographql.com/api/ingress/traces`, I guess most likely because it is not reusing the HTTP contention, so I had to disable the reporting as I could not find a way to pass an HTTP agent to the plunging to keep alive the connection to avoid the issue mentioned above.

Is there a workaround or you guys can please point me to the right direction?

### Link to Reproduction

N/A

**Comment by vany0114** [comment]
Hi @glasser is this repo active/being maintained?

**Comment by glasser** [comment]
Yes, but not by me personally, and most of Apollo was on vacation starting right around when you filed the issue :)
Also, this is more specifically a part of Apollo Server rather than federation-specific, so I will move this issue to that repository. That said, I will then close it for now because there's not enough information to reproduce — at the very least we will need the version of Apollo Server you're using.
